client.go

```diff
@@ -415,6 +414,53 @@ func (d *Dialer) DialContext(ctx context.Context, urlStr string, requestHeader h
 	return conn, resp, nil
 }
 
+// Returns an updated proxy dial function, wrapping the passed
+// "netDial" function, or an error if one occurred. If the proxy
+// is HTTPS, then TLS handshake will occur in the returned dial function.
+func (d *Dialer) proxyNetDial(req *http.Request, netDial netDialerFunc) (netDialerFunc, error) {
+	if d.Proxy == nil {
+		return nil, errors.New("proxy url function is nil")
+	}
+	// Generate the proxy URL and validate.
+	proxyURL, err := d.Proxy(req)
+	if err != nil {
+		return nil, err
+	}
+	if proxyURL == nil {
+		return nil, errors.New("generated proxy url is nil")
+	}
```

Comment:
IIUIC this change the behavior, before if the returned `proxyURL` was nil meant "this connections should not be proxied" and now it will error , I think this should be just a noop
```suggestion
        // this request should not be proxied and use the original dialer
	if proxyURL == nil {
		return netDial, nil
	}
```

Reply:
Fixed.